jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/RFC6265SetCookieParser.java

```diff
@@ -0,0 +1,186 @@
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.http;
+
+/**
+ * <p>A parser for {@code Set-Cookie} header values following
+ * <a href="https://datatracker.ietf.org/doc/html/rfc6265">RFC 6265</a>.</p>
+ * <p>White spaces around cookie name and value, and around attribute
+ * name and value, are permitted but stripped.
+ * Cookie values and attribute values may be quoted with double-quotes.</p>
+ */
+public class RFC6265SetCookieParser implements SetCookieParser
+{
+    @Override
+    public HttpCookie parse(String setCookieValue)
+    {
+        // Implementation of the algorithm from RFC 6265, section 5.2.
+
+        // Parser state.
+        State state = State.NAME;
+        String name = null;
+        boolean quoted = false;
+        HttpCookie.Builder cookie = null;
+        int offset = 0;
+        int length = setCookieValue.length();
+
+        // Parse.
+        for (int i = 0; i < length; ++i)
+        {
+            char ch = setCookieValue.charAt(i);
+            switch (state)
+            {
+                case NAME ->
+                {
+                    HttpTokens.Token token = HttpTokens.getToken(ch);
+                    if (token == null)
+                        return null;
+                    if (ch == '=')
+                    {
+                        name = setCookieValue.substring(offset, i).trim();
+                        if (name.isEmpty())
+                            return null;
+                        offset = i + 1;
+                        state = State.VALUE_START;
+                    }
+                }
+                case VALUE_START ->
+                {
+                    if (Character.isWhitespace(ch))
+                        continue;
+                    if (ch == '"')
+                        quoted = true;
+                    else
+                        --i;
+                    offset = i + 1;
+                    state = State.VALUE;
+                }
+                case VALUE ->
+                {
+                    if (quoted && ch == '"')
+                    {
+                        quoted = false;
+                        String value = setCookieValue.substring(offset, i).trim();
+                        cookie = HttpCookie.build(name, value);
+                        offset = i + 1;
+                        state = State.ATTRIBUTE;
+                    }
+                    else
+                    {
+                        if (ch == ';')
+                        {
+                            String value = setCookieValue.substring(offset, i).trim();
+                            cookie = HttpCookie.build(name, value);
+                            offset = i + 1;
+                            state = State.ATTRIBUTE_NAME;
+                        }
+                    }
```

Comment:
Do you need handling for training white space after the value and before the ; ?

Reply:
Oh I see that is being done with a trim.  Not the most efficient, but OK if spaces before ; are rare.